src/WebGLAtlasTexture.ts

```diff
@@ -457,13 +513,26 @@ export default class WebGLAtlasTexture extends Texture {
       return;
     }
 
-    const [layerIdx, atlasIdx] = textureInfo.id;
+    const gl = this.renderer.context as WebGL2RenderingContext;
 
+    const [layerIdx, atlasIdx] = textureInfo.id;
     const layer = this.layers[layerIdx];
 
-    this.canvas.width = this.canvas.height = layer.size;
-    this.canvasCtx.clearRect(0, 0, layer.size, layer.size);
-    this.uploadImage(layerIdx, atlasIdx, this.canvas);
+    const mips = this.mipFramebuffers[layerIdx];
```

Comment:
Minor, but this code is really close to what we are doing in `addColorRect` and might be worth drying up.